crates/matrix-sdk/src/attachment.rs

```diff
@@ -390,11 +376,17 @@ pub fn generate_image_thumbnail<R: BufRead + Seek>(
     let thumbnail = image.thumbnail(width, height);
     let (thumbnail_width, thumbnail_height) = thumbnail.dimensions();
 
+    let thumbnail_format = match format {
+        ThumbnailFormat::Always(format) => format,
+        ThumbnailFormat::Fallback(format) if !image_format.writing_enabled() => format,
+        _ => image_format,
```

Comment:
Can you use the explicit last variant instead of `_` here, so that if we add a new variant to `ThumbnailFormat`, the compiler warns us here, and we can then decide what's the best thing to do?

Reply:
Changed